packages/api/src/datalayer/links/index.ts

```diff
@@ -444,12 +449,27 @@ class UserArticleModel extends DataModel<
     }
 
     // search by labels using lowercase
-    if (args.labelFilters) {
-      queryPromise
-        .innerJoin(Table.LINK_LABELS, 'link_labels.link_id', 'links.id')
-        .innerJoin(Table.LABELS, 'labels.id', 'link_labels.label_id')
-        .whereRaw('LOWER(omnivore.labels.name) = ANY(?)', [args.labelFilters])
-        .distinct('links.id')
+    if (args.labelFilters.length > 0) {
+      // combine all label filters into one query
+      args.labelFilters.forEach((labelFilter) => {
+        // create a sub query to get the ids of the articles that have the label
+        const subQuery = tx(Table.LINK_LABELS)
+          .select('link_id')
+          .innerJoin(Table.LABELS, 'labels.id', 'link_labels.label_id')
+          .whereRaw('LOWER(omnivore.labels.name) = ANY (?)', [
+            labelFilter.labels,
+          ])
+
+        // filter by include/exclude the result of sub query
+        switch (labelFilter.type) {
+          case LabelFilterType.ANY:
+            queryPromise.whereIn('links.id', subQuery)
+            break
+          case LabelFilterType.NONE:
+            queryPromise.whereNotIn('links.id', subQuery)
+            break
+        }
+      })
```

Comment:
@jacksonh I wrote this `subquery` to support labels search options and it could be improved a bit by using `join`. But still it would be quite inefficient in terms of performance, so I have closed this PR.

Should I start migrating to `elastic` since we need some search options now?

Reply:
Think this one is obsolete now.